tools/kcli/kcli/utils.py

```diff
@@ -0,0 +1,230 @@
+        elif isinstance(key, int):
+            key = str(key)
+
+    if keys:
+        return dict_lookup(dic.get(key, {}), *keys)
+
+    try:
+        value = dic[key]
+    except KeyError:
+        raise exceptions.IRKeyNotFoundException(key, dic)
+
+    LOG.debug("value has been found: \"%s\"" % value)
+    return value
+
+
+def dict_insert(dic, val, key, *keys):
+    """insert a value of a nested key into a dictionary
+
+    to insert value for a nested key, all ancestor keys should be given as
+    method's arguments
+
+    example:
+      dict_lookup({}, 'val', 'key1.key2'.split('.'))
+
+    :param dic: a dictionary object to insert the nested key value into
+    :param val: a value to insert to the given dictionary
+    :param key: first key in a chain of key that will store the value
+    :param keys: sub keys in the keys chain
+    """
+    if not keys:
+        dic[key] = val
+        return
+
+    if key not in dic:
+        dic[key] = {}
+
+    dict_insert(dic[key], val, *keys)
+
+
+# TODO: remove "settings" references in project
+def validate_settings_dir(settings_dir=None):
+    """Checks & returns the full path to the settings dir.
+
+    Path is set in the following priority:
+    1. Method argument
+    2. System environment variable
+    3. Settings dir in the current working dir
+
+    :param settings_dir: path given as argument by a user
+    :return: path to settings dir (str)
+    :raise: IRFileNotFoundException: when the path to the settings dir doesn't
+            exist
+    """
+    settings_dir = settings_dir or os.environ.get(
+        'KHALEESI_SETTINGS') or os.path.join(os.getcwd(), "settings", "")
+
+    if not os.path.exists(settings_dir):
+        raise exceptions.IRFileNotFoundException(
+            settings_dir,
+            "Settings dir doesn't exist: ")
+
+    return settings_dir
+
+
+def merge_settings(settings, file_path):
+    """merge settings in 'file_path' with 'settings'
+
+    :param settings: settings to be merge with
+    :param file_path: path to file with settings to be merged
+    :return: merged settings
+    """
+    LOG.debug("Loading setting file: %s" % file_path)
+    if not os.path.exists(file_path):
+        raise exceptions.IRFileNotFoundException(file_path)
+
+    loaded_file = configure.Configuration.from_file(file_path).configure()
+    settings = settings.merge(loaded_file)
+
+    return settings
+
+
+def generate_settings_file(settings_files, extra_vars):
+    """Generate one settings file from a given list of settings files and
+    extra-vars
+
+    :param settings_files: list of paths to settings files
+    :param extra_vars: list of extra-vars
+    :return: dictionary with merging results of all settings file and
+    extra-vars
+    """
+    settings = configure.Configuration.from_dict({})
+
+    for settings_file in settings_files:
+        settings = merge_settings(settings, settings_file)
+
+    for extra_var in extra_vars:
+        if extra_var.startswith('@'):
```

Comment:
`extra-vars` accepts files as well? Is this doc'ed in "help"? How is it different than the files in "input"?